test/common/stats/thread_local_store_test.cc

```diff
@@ -189,6 +264,157 @@ TEST_F(StatsThreadLocalStoreTest, BasicScope) {
   // Includes overflow stat.
   EXPECT_CALL(*this, free(_)).Times(5);
 }
+TEST_F(StatsThreadLocalStoreTest, BasicSingleHistogramMerge) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Histogram& h1 = store_->histogram("h1");
+  EXPECT_EQ("h1", h1.name());
+
+  expectCallAndAccumulate(h1, 0);
+  expectCallAndAccumulate(h1, 43);
+  expectCallAndAccumulate(h1, 41);
+  expectCallAndAccumulate(h1, 415);
+  expectCallAndAccumulate(h1, 2201);
+  expectCallAndAccumulate(h1, 3201);
+  expectCallAndAccumulate(h1, 125);
+  expectCallAndAccumulate(h1, 13);
+
+  EXPECT_EQ(1, validateMerge());
+
+  // Validate the summary format is as expected.
+  std::string expected_summary = "P0: 0, P25: 14, P50: 44, P75: 420, P90: 3220, P95: 3260, P99: "
+                                 "3292, P99.9: 3299.2, P100: 3300";
```

Comment:
I'm not sure I understand the math here.  None of the inputs are over 3200; why is P100=3300?

Can you add a simpler example and/or explain in a code comment why the value is that way?

Reply:
I have added a simple test case now and validated that the quantile outs are as per the test [here](https://github.com/circonus-labs/libcircllhist/blob/master/src/test/histogram_test.c#L500) .

@postwait can help with the maths aspect here?